### PR TITLE
More 21w44a stuff

### PR DESCRIPTION
**Pull request by apple502j**
JFR, animal spawning, etc.

### CODE REVIEW

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping [hunk 1]
```diff
@@ -136,6 +136,20 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
 		ARG 0 chunk
 	METHOD method_33706 toLong (Lnet/minecraft/class_2338;)J
 		ARG 0 pos
+	METHOD method_39489 forEachChunkSectionsAround (IIILit/unimi/dsi/fastutil/longs/LongConsumer;)V
+		COMMENT Calls {@param consumer} with the chunk section positions one block away
+		COMMENT from the coordinates {@param x}, {@param y}, {@param z} in each direction.
```
Comment (enbrain) [inline mappings/net/minecraft/util/math/ChunkSectionPos.mapping]: Is there a chance this method supplies the same chunk section position multiple times? Might be worth clarifying.

On a side note, `@param` cannot be used as an inline tag.

File: mappings/net/minecraft/util/profiling/jfr/JfrProfiler.mapping [hunk 2]
```diff
@@ -8,12 +8,18 @@ CLASS net/minecraft/class_6505 net/minecraft/util/profiling/jfr/JfrProfiler
 	FIELD field_34409 CONFIG_PATH Ljava/lang/String;
 	FIELD field_34410 DATE_TIME_FORMAT Ljava/time/format/DateTimeFormatter;
 	FIELD field_34411 currentRecording Ljdk/jfr/Recording;
+	FIELD field_35576 JFR_PROFILER_INSTANCE Lnet/minecraft/class_6505;
```
Comment (enbrain) [inline mappings/net/minecraft/util/profiling/jfr/JfrProfiler.mapping]: Just `INSTANCE`?
```suggestion
	FIELD field_35576 INSTANCE Lnet/minecraft/class_6505;
```

File: mappings/net/minecraft/util/profiling/jfr/event/NetworkSummaryEvent.mapping [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+		FIELD field_35644 RECEIVED_BYTES Ljava/lang/String;
+		FIELD field_35645 SENT_PACKETS Ljava/lang/String;
+		FIELD field_35646 RECEIVED_PACKETS Ljava/lang/String;
+	CLASS class_6779 Recorder
+		FIELD field_35647 sentBytes Ljava/util/concurrent/atomic/AtomicLong;
+		FIELD field_35648 sentPackets Ljava/util/concurrent/atomic/AtomicInteger;
+		FIELD field_35649 receivedBytes Ljava/util/concurrent/atomic/AtomicLong;
+		FIELD field_35650 receivedPackets Ljava/util/concurrent/atomic/AtomicInteger;
+		FIELD field_35651 event Lnet/minecraft/util/profiling/jfr/event/NetworkSummaryEvent;
+		METHOD <init> Recorder (Ljava/lang/String;)V
```
Comment (enbrain) [inline mappings/net/minecraft/util/profiling/jfr/event/NetworkSummaryEvent.mapping]:
```suggestion
		METHOD <init> (Ljava/lang/String;)V
```
Did Enigma allow it?

File: mappings/net/minecraft/util/profiling/jfr/FlightProfiler.mapping [hunk 4]
```diff
@@ -11,7 +11,15 @@ CLASS net/minecraft/class_6611 net/minecraft/util/profiling/jfr/FlightProfiler
 		ARG 2 world
 		ARG 3 targetStatus
 	METHOD method_38656 onPacketReceived (IILjava/net/SocketAddress;I)V
+		ARG 1 stateId
```
Comment (enbrain) [inline mappings/net/minecraft/util/profiling/jfr/FlightProfiler.mapping]:
```suggestion
		ARG 1 protocolId
```
For consistency with `PacketEvent#protocolId`.

File: mappings/net/minecraft/util/profiling/jfr/FlightProfiler.mapping [hunk 5]
```diff
@@ -11,7 +11,15 @@ CLASS net/minecraft/class_6611 net/minecraft/util/profiling/jfr/FlightProfiler
...
+		ARG 2 packetId
+		ARG 3 address
+		ARG 4 bytes
 	METHOD method_38657 onPacketSent (IILjava/net/SocketAddress;I)V
+		ARG 1 stateId
```
Comment (enbrain) [inline mappings/net/minecraft/util/profiling/jfr/FlightProfiler.mapping]:
```suggestion
		ARG 1 protocolId
```

File: mappings/net/minecraft/util/profiling/jfr/JfrProfileRecorder.mapping [hunk 6]
```diff
@@ -3,8 +3,8 @@ CLASS net/minecraft/class_6516 net/minecraft/util/profiling/jfr/JfrProfileRecord
 	FIELD field_34423 endTime Ljava/time/Instant;
 	FIELD field_34424 chunkGenerationSamples Ljava/util/List;
 	FIELD field_34425 cpuLoadSamples Ljava/util/List;
-	FIELD field_34426 packetReadSamples Ljava/util/Map;
-	FIELD field_34427 packetSentSamples Ljava/util/Map;
+	FIELD field_34426 c2sPacketsToCounter Ljava/util/Map;
+	FIELD field_34427 s2cPacketsToCounter Ljava/util/Map;
```
Comment (enbrain) [inline mappings/net/minecraft/util/profiling/jfr/JfrProfileRecorder.mapping]: Doesn't JFR work on both sides?

File: mappings/net/minecraft/util/profiling/jfr/JfrProfiler.mapping (hunk 2 above)
Comment (apple502j) [reply mappings/net/minecraft/util/profiling/jfr/JfrProfiler.mapping]: Its superclass has a public field currently named `INSTANCE`.

File: mappings/net/minecraft/util/profiling/jfr/event/NetworkSummaryEvent.mapping (hunk 3 above)
Comment (apple502j) [reply mappings/net/minecraft/util/profiling/jfr/event/NetworkSummaryEvent.mapping]: Yeah, probably a Enigma bug.

File: mappings/net/minecraft/util/profiling/jfr/JfrProfiler.mapping (hunk 2 above)
Comment (enbrain) [reply mappings/net/minecraft/util/profiling/jfr/JfrProfiler.mapping]: That's a known Enigma bug (https://github.com/FabricMC/Enigma/issues/191). I'd assume it's safe to rename it manually.

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping (hunk 1 above)
Comment (enbrain) [reply mappings/net/minecraft/util/math/ChunkSectionPos.mapping]: Looks like it never supplies the same chunk section multiple times. How about this?

```suggestion
	METHOD method_39489 forEachChunkSectionsAround (IIILit/unimi/dsi/fastutil/longs/LongConsumer;)V
		COMMENT Performs an action for each chunk section enclosing a block position
		COMMENT adjacent to {@code (x, y, z)}.
```

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping [hunk 7]
```diff
@@ -136,6 +136,20 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
 		ARG 0 chunk
 	METHOD method_33706 toLong (Lnet/minecraft/class_2338;)J
 		ARG 0 pos
+	METHOD method_39489 forEachChunkSectionsAround (IIILit/unimi/dsi/fastutil/longs/LongConsumer;)V
```
Comment (enbrain) [inline mappings/net/minecraft/util/math/ChunkSectionPos.mapping]:
```suggestion
	METHOD method_39489 forEachChunkSectionAround (IIILit/unimi/dsi/fastutil/longs/LongConsumer;)V
```

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping [hunk 8]
```diff
@@ -136,6 +136,20 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
...
+		COMMENT Performs an action for each chunk section enclosing a block position
+		COMMENT adjacent to {@code (x, y, z)}.
+		ARG 0 x
+		ARG 1 y
+		ARG 2 z
+		ARG 3 consumer
+			COMMENT the consumer that takes the chunk section position as a long
+	METHOD method_39490 forEachChunkSectionsAround (JLit/unimi/dsi/fastutil/longs/LongConsumer;)V
```
Comment (enbrain) [inline mappings/net/minecraft/util/math/ChunkSectionPos.mapping]:
```suggestion
	METHOD method_39490 forEachChunkSectionAround (JLit/unimi/dsi/fastutil/longs/LongConsumer;)V
```

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping [hunk 9]
```diff
@@ -136,6 +136,20 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
...
+		ARG 0 pos
+		ARG 2 consumer
+	METHOD method_39491 forEachChunkSectionsAround (Lnet/minecraft/class_2338;Lit/unimi/dsi/fastutil/longs/LongConsumer;)V
```
Comment (enbrain) [inline mappings/net/minecraft/util/math/ChunkSectionPos.mapping]:
```suggestion
	METHOD method_39491 forEachChunkSectionAround (Lnet/minecraft/class_2338;Lit/unimi/dsi/fastutil/longs/LongConsumer;)V
```

File: mappings/net/minecraft/util/math/ChunkSectionPos.mapping [hunk 10]
```diff
@@ -136,6 +136,20 @@ CLASS net/minecraft/class_4076 net/minecraft/util/math/ChunkSectionPos
 		ARG 0 chunk
 	METHOD method_33706 toLong (Lnet/minecraft/class_2338;)J
 		ARG 0 pos
+	METHOD method_39489 forEachChunkSectionsAround (IIILit/unimi/dsi/fastutil/longs/LongConsumer;)V
+		COMMENT Performs an action for each chunk section enclosing a block position
+		COMMENT adjacent to {@code (x, y, z)}.
```
Comment (enbrain) [inline mappings/net/minecraft/util/math/ChunkSectionPos.mapping:141]:
```suggestion
		COMMENT adjacent or equal to {@code (x, y, z)}.
```

Comment (enbrain) [reply mappings/net/minecraft/util/math/ChunkSectionPos.mapping:141]: On a second thought, just adjacent suffices.

Comment (apple502j) [reply mappings/net/minecraft/util/math/ChunkSectionPos.mapping:141]: Don't think it checked for that exact coords. But shouldn't matter, chunk section is 16x16x16.

**Review by enbrain — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**